spec/abstractops.html

```diff
@@ -1428,12 +1427,15 @@ <h1>
       <dd>It parses the argument as an ISO 8601 string and returns the calendar identifier, or *undefined* if there is none.</dd>
     </dl>
     <emu-alg>
-      1. Let _parseResult_ be ParseText(StringToCodePoints(_isoString_), |TemporalCalendarString|).
+      1. Let _parseResult_ be ParseText(StringToCodePoints(_isoString_), |CalendarDateTime|).
       1. If _parseResult_ is a List of errors, throw a *RangeError* exception.
```

Comment:
I think we want to move this line down, after the second ParseText call. We don't want to throw at this point if the string is not a valid CalendarDateTime - that would throw on e.g. `Temporal.Calendar.from("iso8601")`.

Instead here we should only check the |Calendar| Parse Node if _parseResult_ is itself a Parse Node, and if it's a List of errors, we should skip to the next ParseText call. You could take the opportunity to combine and rewrite that as something like

1. If _parseResult_ is a Parse Node that contains a |CalendarName| Parse Node, then
  - 1. Let _calendar_ be the source text matched by the |CalendarName| Parse Node contained within _parseResult_.
  - 2. Return CodePointsToString(_calendar_).